commit 4b8adabf8d6d821ad6fde20588a5ef0d72363a97
Author: Christian Mollekopf <chrigi_1@fastmail.fm>
Date:   Tue Nov 19 18:29:27 2013 +0100

    Always reset doSslFallback.
    
    Otherwise doSslFallback will remain true if the errors are ignored, although a connection is established.
    If we then receive an error from the server (e.g. due to disconnect), the session
    will go into a reconnect frenzy.
    
    BUG: 316840
    FIXED-IN: 4.11.4

diff --git a/kimap/sessionthread.cpp b/kimap/sessionthread.cpp
index e70e317..42a1f62 100644
--- a/kimap/sessionthread.cpp
+++ b/kimap/sessionthread.cpp
@@ -274,6 +274,7 @@ void SessionThread::socketError(KTcpSocket::Error error)
 // Called in secondary thread
 void SessionThread::sslConnected()
 {
+  doSslFallback = false;
   Q_ASSERT( QThread::currentThread() == thread() );
   if ( !m_socket )
     return;
@@ -290,7 +291,6 @@ void SessionThread::sslConnected()
      KSslErrorUiData errorData( m_socket );
      emit sslError( errorData );
   } else {
-    doSslFallback = false;
     kDebug() << "TLS negotiation done.";
     m_encryptedMode = true;
     emit encryptionNegotiationResult( true, m_socket->negotiatedSslVersion() );
